pkg/binding/bindingpolicy-resolution.go

```diff
@@ -184,38 +215,45 @@ func destinationsMatch(resolvedDestinations sets.Set[string], bindingDestination
 	return true
 }
 
-type objectRefAndVersion struct {
+type objectRefAndData struct {
 	schema.GroupVersionResource
 	cache.ObjectName
-	ResourceVersion string
+
+	*objectData
 }
 
-func bindingObjectRefAndVersionSetFromWorkload(bindingSpecWorkload *v1alpha1.DownsyncObjectReferences) sets.Set[objectRefAndVersion] {
-	bindingObjectRefAndVersionSet := sets.New[objectRefAndVersion]()
+func bindingObjectRefAndDataSetFromWorkload(bindingSpecWorkload *v1alpha1.DownsyncObjectReferences) sets.Set[objectRefAndData] {
+	bindingObjectRefAndDataSet := sets.New[objectRefAndData]()
 
 	for _, clusterScopeDownsyncObject := range bindingSpecWorkload.ClusterScope {
-		bindingObjectRefAndVersionSet.Insert(objectRefAndVersion{
+		bindingObjectRefAndDataSet.Insert(objectRefAndData{
 			GroupVersionResource: schema.GroupVersionResource(clusterScopeDownsyncObject.GroupVersionResource),
 			ObjectName: cache.ObjectName{
 				Name:      clusterScopeDownsyncObject.Name,
 				Namespace: metav1.NamespaceNone,
 			},
-			ResourceVersion: clusterScopeDownsyncObject.ResourceVersion,
+			objectData: &objectData{
+				ResourceVersion:  clusterScopeDownsyncObject.ResourceVersion,
+				StatusCollectors: sets.New[string](clusterScopeDownsyncObject.StatusCollectors...),
```

Comment:
FYI, I think that you can s/[string]// here, Go's type inference will figure it out.